security/postmortems/cve-2019-15225.md

```diff
@@ -0,0 +1,138 @@
+# Security postmortem for CVE-2019-15225, CVE-2019-15226
+
+## Incident date(s)
+
+2019-07-25 - 2019-10-10
+
+## Authors
+
+@asraa
+
+## Status
+
+Final
+
+## Summary
+
+After an Envoy user publicly reported a crash in Envoy about regular expression matching in route resolution (https://github.com/envoyproxy/envoy/issues/7728), the Envoy security team found that issue could be leveraged for a DoS attack and would go through the public security release process. The fix landed in master with a public PR (https://github.com/envoyproxy/envoy/pull/7878) and was targeted to be included in a 1.11.2 security release.
+
+CVE-2019-15226 was detected via fuzzers just after the 1.11.1 security release. With the fix of CVE-2019-15225 in progress, the Envoy security team decided to lump the two fixes into a 1.11.2 security release. This was the first time in which the Envoy security release included a publicly disclosed vulnerability with a fix that was merged into master. The security release included a backported patch of the fix as well as the patches for CVE-2019-15226.
+
+## CVE issue(s)
+
+* https://github.com/envoyproxy/envoy/issues/8519
+* https://github.com/envoyproxy/envoy/issues/8520
+
+## Root Causes
+
+CVE-2019-15225 was caused by the use of a recursive algorithm for matching regular expressions. Envoy’s HTTP router can be configured with regular expressions for routing incoming HTTP requests that matched header values. Envoy used the libstdc++ std::regex implementation for these regular expressions. As a result, an HTTP request with sufficiently large header values may consume large amounts of stack memory and cause abnormal process termination. Regular expressions with the `*` or `+` quantifiers are particularly vulnerable and may cause abnormal process termination. This appeared when matching header values of 16Kb or more.
```

Comment:
Nit: can you either use backticks for `std::regex` formatting or Markdown link to the STL docs?